### PR TITLE
The design of front end

### PR DESCRIPTION
**Pull request by Annazhan**
The detail of design is shown in frontEnd-Design.md

### CODE REVIEW

**Comment by yalhessi** [comment]
Nice work! I like what you're proposing, and I think it would be nice to incorporate a few things into your proposal.

1) It seems like the tasks you're proposing fall under a few categories, maybe something like: html page layout, some minor ide-like features, some debugging. I'm not sure which exact buckets make the most sense, so I'd love to hear from you about that. You're mostly focused on UI/UX so it's useful to specify ahead of time which part of the user experience you're trying to improve. 

2) When it comes to proposing changes to UI, it's really useful to give us a sense of what the new look is visually. There's many ways frontend developers tend to do this, like breadboarding, fatmarkers, mockups, etc. Basically anything that's visual so we can have a better sense of what you're trying to accomplish.

**Comment by Annazhan** [comment]
Front end week 7

File: .vscode/dryrun.log [hunk 1]
```diff
@@ -0,0 +1,7 @@
+make.exe --dry-run --always-make --keep-going --print-directory
```
Comment (GV1028) [inline .vscode/dryrun.log]: Just making sure, do we need dryrun.log and targets.log or should they be in .gitignore?

File: repl.ts [hunk 2]
```diff
@@ -15,6 +22,7 @@ export class BasicREPL {
   functions: string
   importObject: any
   memory: any
+
```
Comment (GV1028) [inline repl.ts:25]: Clean up whitespace/newline

File: repl.ts [hunk 3]
```diff
@@ -33,6 +41,8 @@ export class BasicREPL {
     this.currentTypeEnv = defaultTypeEnv;
     this.functions = "";
   }
+
```
Comment (GV1028) [inline repl.ts:44]: Clean up whitespace/newline

File: test.py [hunk 4]
```diff
@@ -0,0 +1,3 @@
+print(1)
```
Comment (GV1028) [inline test.py]: What is test.py used for? Is it just for your debugging? In the future, please do explain all the changes you made to each file in your design.md

File: webstart.ts [hunk 5]
```diff
@@ -1,96 +1,66 @@
...
+import "codemirror/addon/fold/comment-fold";
+import "./style.scss";
 
 function assert_not_none(arg: any) : any {
   if (arg === 0)
     throw new Error("RUNTIME ERROR: cannot perform operation on none");
   return arg;
 }
 
+console.log('this is ok')
```
Comment (GV1028) [inline webstart.ts]: Clean up on all console.log or other debugging statements

**Review by GV1028 — COMMENTED** [review]
From your design, looks like you have solved quite a few issues in the existing UI. Its hard to visualize the changes without any screenshots of how the UI looks. Also, it would be nice to see test cases to demonstrate different functionalities of UI with screenshots, and your efforts to make sure there are no bugs.

File: test.py (hunk 4 above)
Comment (chengh233) [reply test.py]: Thank you for your comment. It is just for debugging and it has been removed. Sorry for the confusion!

**Comment by Annazhan** [comment]
Hi! Thanks for your time and comments! Our team have already deleted all the debug file log file and other debug statements.  However, I still have questions about how can we upload our UI with screen shots, should we upload in comment or should we push to our repository?

File: .vscode/dryrun.log (hunk 1 above)
Comment (Gilone) [reply .vscode/dryrun.log]: Thank you! We have added .vscode/*.log in .gitignore.

File: repl.ts (hunk 2 above)
Comment (Gilone) [reply repl.ts:25]: Done, thank you!

File: repl.ts (hunk 3 above)
Comment (Gilone) [reply repl.ts:44]: Done, thank you!

File: webstart.ts (hunk 5 above)
Comment (Gilone) [reply webstart.ts]: Thanks for the reminder! 
Yes, leaving debugging statements in a public repo is a bad coding habit.

**Pull request merged**